From dd557127f554fc65f1d503585091bc182fd4ceb2 Mon Sep 17 00:00:00 2001
From: Luiz Augusto von Dentz <luiz.dentz@openbossa.org>
Date: Tue, 25 Nov 2008 17:20:45 -0300
Subject: [PATCH] Do not unload module-bluetooth-device when io thread fails.

Let module-bluetooth-discover decide when to unload device module,
otherwise modules may be unloading twice causing invalid reads.
---
 src/modules/bluetooth/module-bluetooth-device.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/src/modules/bluetooth/module-bluetooth-device.c b/src/modules/bluetooth/module-bluetooth-device.c
index cb4746a..bf0caa2 100644
--- a/src/modules/bluetooth/module-bluetooth-device.c
+++ b/src/modules/bluetooth/module-bluetooth-device.c
@@ -779,7 +779,6 @@ static void thread_func(void *userdata) {
 fail:
     /* If this was no regular exit from the loop we have to continue processing messages until we receive PA_MESSAGE_SHUTDOWN */
     pa_log_debug("IO thread failed");
-    pa_asyncmsgq_post(u->thread_mq.outq, PA_MSGOBJECT(u->core), PA_CORE_MESSAGE_UNLOAD_MODULE, u->module, 0, NULL, NULL);
     pa_asyncmsgq_wait_for(u->thread_mq.inq, PA_MESSAGE_SHUTDOWN);
 
 finish:
-- 
1.6.0.2.514.g23abd3

